pkg/test/nodeclass.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+
+	"github.com/awslabs/operatorpkg/status"
+
+	"sigs.k8s.io/karpenter/pkg/test/v1test1"
+
+	"github.com/imdario/mergo"
+)
+
+// NodeClass creates a test NodeClass with defaults that can be overridden by overrides.
+// Overrides are applied in order, with a last write wins semantic.
+func NodeClass(overrides ...v1test1.TestNodeClass) *v1test1.TestNodeClass {
```

Comment:
```suggestion
func TestNodeClass(overrides ...v1test1.TestNodeClass) *v1test1.TestNodeClass {
```

Reply:
Only to match the name

Reply:
Kept it as NodeClass because of the warning that function name should not start with package name.